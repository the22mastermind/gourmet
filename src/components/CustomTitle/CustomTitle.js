import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import { Title } from 'react-native-paper';

const styles = StyleSheet.create({
  title: {
    fontSize: 24,
  },
});

const CustomTitle = ({ text }) => (
  <Title testID="title" accessibilityLabel="title" style={styles.title}>
    {text}
  </Title>
);

CustomTitle.propTypes = {
  text: PropTypes.string.isRequired,
};

export default CustomTitle;
