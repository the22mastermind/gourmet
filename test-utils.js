import React from 'react';
import { render } from '@testing-library/react-native';
import PaperProvider from 'react-native-paper/lib/commonjs/core/Provider';

const AllTheProviders = ({ children }) => (
  <PaperProvider>{children}</PaperProvider>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
