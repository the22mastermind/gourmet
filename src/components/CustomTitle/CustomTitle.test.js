import React from 'react';
import { render } from 'test-utils';
import CustomTitle from './CustomTitle';

describe('<CustomTitle /> component', () => {
  it('renders <CustomTitle /> correctly', () => {
    const screen = render(<CustomTitle text="This is a title" />);
  
    expect(screen.getByTestId('title')).toBeTruthy();
    expect(screen.queryByText('This is a title')).toBeTruthy();
  });
  
  it('matches <CustomTitle /> component snapshot', async () => {
    const screen = render(<CustomTitle text="This is a title" />);
  
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
