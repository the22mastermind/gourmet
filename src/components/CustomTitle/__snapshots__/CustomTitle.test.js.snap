// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTitle /> component matches <CustomTitle /> component snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessibilityLabel="title"
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "rgba(51, 51, 51, 0.87)",
            "fontFamily": "System",
            "fontWeight": "500",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "fontSize": 24,
            },
          ],
        ],
      ]
    }
    testID="title"
  >
    This is a title
  </Text>
</View>
`;
